lib/extensions/ar_adapter/ar_pglogical.rb

```diff
@@ -0,0 +1,228 @@
+    raw_connection.exec_params("SELECT pglogical.alter_replication_set($1::VARCHAR, $2::BOOL, $3::BOOL, $4::BOOL, $5::BOOL)",
+                      [set_name, insert, update, delete, truncate])
+  end
+
+  # Removes a replication set
+  #
+  # @param set_name [string] replication set name
+  def pglogical_replication_set_drop(set_name)
+    raw_connection.exec_params("SELECT pglogical.drop_replication_set($1::VARCHAR)", [set_name])
+  end
+
+  # Adds a table to a replication set
+  #
+  # @param set_name [String] replication set name
+  # @param table_name [String] table name to add
+  # @param sync [Boolean] sync the table on all subscribers to the given replication set
+  def pglogical_replication_set_add_table(set_name, table_name, sync = false)
+    raw_connection.exec_params("SELECT pglogical.replication_set_add_table($1::VARCHAR, $2::VARCHAR, $3::BOOL)",
+                      [set_name, table_name, sync])
+  end
+
+  # Adds all tables in the given schemas to the replication set
+  #
+  # @param set_name [String] replication set name
+  # @param schema_names [Array<String>] list of schema names
+  # @param sync [Boolean] sync table data to all the subscribers to the replication set
+  def pglogical_replication_set_add_all_tables(set_name, schema_names, sync = false)
+    p_num = 1
+    command = "SELECT pglogical.replication_set_add_all_tables("
+    # set_name
+    command << "$#{p_num}::VARCHAR,"
+    p_num += 1
+    # schema_names
+    command << pglogical_array_param(p_num, schema_names.size, "VARCHAR")
+    p_num += schema_names.size
+    # sync
+    command << ",$#{p_num}::BOOL)"
+
+    raw_connection.exec_params(command, [set_name, *schema_names, sync])
+  end
+
+  # Removes a table from a replication set
+  #
+  # @param set_name [String] replication set name
+  # @param table_name [String] table to remove
+  def pglogical_replication_set_remove_table(set_name, table_name)
+    raw_connection.exec_params("SELECT pglogical.replication_set_remove_table($1::VARCHAR, $2::VARCHAR)", [set_name, table_name])
+  end
+
+  private
+
+  # Create a SQL string that represents a typed array for use in `raw_connection.exec_params`
+  #   The parameter placeholders (eg. $2) will start at `param_start`
+  #
+  # @param param_start [Fixnum] number of first parameter placeholder
+  # @param size [Fixnum] number of parameter placeholders in the array
+  # @param type [String] SQL type of the array elements
+  # @return [String] string of the form `ARRAY[$2,$3,$4]::VARCHAR[]`
+  def pglogical_array_param(param_start, size, type)
+    p_num = param_start
+    rep_sets_str = "ARRAY["
+    (size - 1).times do
+      rep_sets_str << "$#{p_num}, "
+      p_num += 1
+    end
+    rep_sets_str << "$#{p_num}]::#{type}[]"
+  end
+end
+
+module ActiveRecord
+  module ConnectionAdapters
+    class PostgreSQLAdapter < AbstractAdapter
+      include PgLogical
```

Comment:
@matthewd Was thinking about this and, personally, I am of the firm opinion that these methods belong on the adapter, for a number of reasons.  However, I agree that the psuedo-namespacing is kind of gross, and we are introducing a *lot* of methods.  What do you think of something like this?

```ruby
class PgLogicalRaw
  attr_accessor :connection

  def initialize(connection)
    @connection = connection
  end

  def installed?; connection.execute("stuff"); end
  def enable; connection.execute("stuff"); end
  # ...
end

module PgLogicalAdapterMixin
  def pglogical
    PgLogicalRaw.new(self)
  end
end

ActiveRecord::ConnectionAdapters::PostgreSQLAdapter.include PgLogicalAdapterMixin
```

So instead of calling `connection.pglogical_installed?`, one would call `connection.pglogical.installed?`